exercises/all-your-base/canonical-data.json

```diff
@@ -92,31 +101,67 @@
       "input_base": 2,
       "input_digits": [],
       "output_base": 10,
-      "expected": null
+      "expected": {
+        "choices": {
+          "if_your_track_allows_empty_input_digits": {
+            "choices": {
+              "if_your_track_translates_empty_to_empty": [],
+              "if_your_track_outputs_zero_as_a_single_zero": [0],
+              "otherwise": []
+            }
+          },
+          "otherwise": {"error": "input digits cannot be empty"}
+        }
+      }
     },
     {
       "description": "single zero",
       "property": "rebase",
       "input_base": 10,
       "input_digits": [0],
       "output_base": 2,
-      "expected": null
+      "expected": {
+        "choices": {
+          "if_your_track_allows_single_zero_as_input": {
+            "choices": {
+              "if_your_track_outputs_zero_as_a_single_zero": [0],
+              "otherwise": []
+            }
+          },
+          "otherwise": {"error": "leading zeroes not allowed"}
+        }
+      }
     },
     {
       "description": "multiple zeros",
       "property": "rebase",
       "input_base": 10,
       "input_digits": [0, 0, 0],
       "output_base": 2,
-      "expected": null
+      "expected": {
+        "choices": {
+          "if_your_track_allows_leading_zeroes_in_input": {
+            "choices": {
```

Comment:
bug: For tracks that allow leading zeroes in **nonzero** inputs BUT expect their zeroes to nonetheless be `[]`, this should be an error, but currently isn't. Oh no, how could I make such a grievous error on such an important PR